build-system/tasks/coverage-map/index.js

```diff
@@ -78,26 +81,46 @@ async function autoScroll(page) {
   });
 }
 
+async function htmlTransform() {
+  log(`Transforming ${inputHtml}...`);
+  const transformed = await transform(`examples/${inputHtml}`);
+  const transformedName = `transformed.${inputHtml}`;
+  await fsExtra.ensureDir('dist/transformed');
+  await fs.writeFile(`dist/transformed/${transformedName}`, transformed);
+  log(
+    `Transformation complete. It can be found at "dist/transformed/${transformedName}".`
+  );
+  testUrl = `http://localhost:${serverPort}/dist/transformed/${transformedName}`;
+}
+
 async function generateMap() {
   log(
     `Generating heat map in dist/${outHtml} of ${inputJs}, based on ${coverageJsonName}...`
   );
   await explore(`dist/${inputJs}`, {
-    output: {format: 'html', filename: `${outHtml}`},
+    output: {format: 'html', filename: `dist/${outHtml}`},
     coverage: `dist/${coverageJsonName}`,
     onlyMapped: true,
   });
 }
 
 async function coverageMap() {
   installPackages(__dirname);
+  buildNewServer();
+
+  if (!argv.nobuild) {
+    await dist();
+  }
 
   puppeteer = require('puppeteer');
   explore = require('source-map-explorer').explore;
+  transform = await require('../../server/new-server/transforms/dist/transform')
+    .transform;
 
-  if (!argv.nobuild) {
-    await dist();
+  if (argv.esm || argv.sxg) {
```

Comment:
Give that right now this is the only place these flags are used, would it make more sense to have a single `--transform` flag instead of these separate flags, and let calling code for esm/sxg variants set that flag?

Reply:
I made the decision to use these flags because the `dist` task also uses the `esm` and `sxg` flags, and I did not want to change `dist.js` unless absolutely necessary... What would be a better way to change this?

Reply:
I'll defer to @rsimha 's opinion here on how to handle variations/overlay in flags (this may be fine as-is), not sure what our convention is

Reply:
In this case, I think @xiexr151e's point about `esm` and `sxg` already being well-established flags across all our tasks is valid. In future, if the distinction between them were to disappear, it may make sense to redefine the flags across the toolchain. Until then, leaving them as-is SGTM. 

PS: Thanks for the comments all around. It's nice to see the thought y'all are putting into usability and maintainability 😃